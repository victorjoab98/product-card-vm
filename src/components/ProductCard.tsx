import  React,{ createContext } from 'react'
import styles from '../styles/styles.module.css'
import { useProduct } from '../hooks'
import { InitialValuesProductCard, onChangeProductArgs, Product, ProductCardHandlers, ProductContextProps } from '../interfaces';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export interface Props {
  product: Product,
  // children?: ReactElement | ReactElement[],
  children: ( args: ProductCardHandlers ) => JSX.Element,
  className?: string,
  style?: React.CSSProperties,
  onChange?: ( args: onChangeProductArgs ) => void,
  value?: number,
  initialValues?: InitialValuesProductCard
}

export const ProductCard = ( { product, children, className, style, onChange, value, initialValues } : Props ) => {

  const { counter, increaseBy, reset, maxCount, isMaxCountReached } = useProduct({ product, onChange, value, initialValues })

  return (
    <Provider value={{
        product,
        counter,
        increaseBy,
        maxCount
    }}> 
        <div className={ `${ styles.productCard } ${ className }` } style={style}>
            { children({
                count: counter,
                isMaxCountReached,
                maxCount,
                product,
                increaseBy,
                reset
            }) }
        </div>
    </Provider>
  )
}
