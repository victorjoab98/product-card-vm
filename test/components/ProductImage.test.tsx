import React from "react"
import rederer from "react-test-renderer"
import ProductCard, { ProductImage } from "../../src/components"
import { product1 } from "../data/products"



describe('ProductImage', () => { 
    test('debe mostrarse con imagen', () => {
        const wrapper = rederer.create(
            <ProductImage img="http://image.jpg"/>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    })

    test('debe mostrarse con imagen del componente padre', () => {
        const wrapper = rederer.create(
            <ProductCard product={product1}>
                {
                    () => <ProductImage/>
                 }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    })
 })