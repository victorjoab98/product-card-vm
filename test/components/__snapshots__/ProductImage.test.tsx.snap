// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrarse con imagen 1`] = `
<img
  alt="Coffe Mug"
  className="productImg undefined"
  src="http://image.jpg"
/>
`;

exports[`ProductImage debe mostrarse con imagen del componente padre 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffe Mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
